### ISSUE TITLE
Don't shows some msgid for English language

### ISSUE THREAD

**Issue by macagua** [smcmahon/Products.PloneFormGen#178]
Hi @smcmahon 

- Branch: 1.7-maintenance
- Plone version: 4.3.x

The default language for Products.PloneFormGen package is English, but enabled this package with English languague at Plone site, don't shows some msgid for English language:

Check out the following images: 

![captura de pantalla de 2016-11-17 20 18 09](https://cloud.githubusercontent.com/assets/185395/20414613/ab373148-ad09-11e6-81e8-a83d6df4db04.png)
![captura de pantalla de 2016-11-17 20 18 05](https://cloud.githubusercontent.com/assets/185395/20414614/ab4e82bc-ad09-11e6-935f-966d521d340b.png)
![captura de pantalla de 2016-11-17 20 17 59](https://cloud.githubusercontent.com/assets/185395/20414615/ab6c97de-ad09-11e6-9cde-a4e5d8973aa1.png)
![captura de pantalla de 2016-11-17 20 17 52](https://cloud.githubusercontent.com/assets/185395/20414616/ab843a10-ad09-11e6-8a7b-415e45492e47.png)
![captura de pantalla de 2016-11-17 20 17 32](https://cloud.githubusercontent.com/assets/185395/20414617/aba80ca6-ad09-11e6-8b3f-ff1211cf5cb6.png)
![captura de pantalla de 2016-11-17 20 17 26](https://cloud.githubusercontent.com/assets/185395/20414618/abbb2e3a-ad09-11e6-8744-7c40d068d2d9.png)
![captura de pantalla de 2016-11-17 20 17 22](https://cloud.githubusercontent.com/assets/185395/20414619/abcf79a8-ad09-11e6-9354-949889fee01f.png)
![captura de pantalla de 2016-11-17 20 17 17](https://cloud.githubusercontent.com/assets/185395/20414620/abe5b38a-ad09-11e6-985f-78b6e1aefa65.png)
![captura de pantalla de 2016-11-17 20 17 04](https://cloud.githubusercontent.com/assets/185395/20414621/abfb8084-ad09-11e6-8d29-4b3d63daeed0.png)

The msgid wrongs more common has values than including characters like "\n", tab space or simple spaces, when this is a simple long strings.

Any Idea @smcmahon ? 

Thanks you!

**Comment by macagua** [comment]
@mauritsvanrees Any idea?

**Comment by mauritsvanrees** [comment]
This is weird.

First, the English locale should not be needed at all. There used to be reasons to add it for cases where you support more than one language in your site, [as I describe in my post here](https://groups.google.com/forum/#!topic/plone-i18n/NMKcFXCTIKI) but as far as I remember this is not a problem anymore in Plone 4, and certainly not in 4.3.

But it's not just happening in English, but also in Dutch.

... dig, dig, dig...

Aha! Our `form.py` has a `widget` method: https://github.com/smcmahon/Products.PloneFormGen/blob/1.7.19/Products/PloneFormGen/content/form.py#L1112
When I remove this method, the translations work again.

This was broken since commit 74f3fa1a2ff44b3362a5cf3671b818fc0a55e95b which added this method to fix a XSS problem.

I am preparing a fix.

**Comment by macagua** [comment]
Hi @smcmahon @mauritsvanrees 

The problem is about the way to add i18n support, check out:
 
![2016-11-18 07 26 31](https://cloud.githubusercontent.com/assets/185395/20429312/41d5df6e-ad64-11e6-88a5-b705a6d54d77.png)

Here my python source code with the changes for this test:

```python
    # ORIGINAL FIELD SCHEMA
    LinesField('actionAdapter',
        vocabulary='actionAdaptersDL',
        widget=MultiSelectionWidget(
            label=_(u'label_actionadapter_text', default=u"Action Adapter"),
            description=_(u'help_actionadapter_text', default=u"""
                To make your form do something useful when submitted:
                add one or more form action adapters to the form folder,
                configure them, then return to this
                form and select the active ones.
                """),
            format='checkbox',
            ),
        ),
    # COPY FIELD SCHEMA
    LinesField('actionAdapterTest',
        vocabulary='actionAdaptersDL',
        widget=MultiSelectionWidget(
            label=_(u'label_actionadapter_text_test', default=u"Action Adapter TEST"),
            description=_(u"To make your form do something useful when submitted: "
                "add one or more form action adapters to the form folder, "
                "configure them, then return to this "
                "form and select the active ones."),
            format='checkbox',
            ),
        ),
```

**Comment by mauritsvanrees** [comment]
Yes, that extra field works because there is no i18n message id. Or rather: the message id is the default text. My pull request fixes it.

**Comment by macagua** [comment]
@mauritsvanrees for me the Spanish translations working very well 
![2016-11-18 08 06 39](https://cloud.githubusercontent.com/assets/185395/20429621/34d39714-ad66-11e6-96bf-442753476e50.png)

**Comment by macagua** [comment]
@mauritsvanrees Yes the problem is the way as defined message id for the translations strings. 

Thanks for your pull request, I need to fixes this issue on a Website for a client. If you need some help for test something just tell me!

**Comment by mauritsvanrees** [comment]
Without my fix, the ugly message ids show up in Dutch for me, both in 4.3 and 5.0. For example when editing an email field:

![screen shot 2016-11-18 at 13 26 31](https://cloud.githubusercontent.com/assets/210587/20430242/860b9c44-ad93-11e6-983d-546bcef1d643.png)

So for me the problem is both in English and in other languages.
I will make a separate pull request for Plone 5.0.

**Comment by macagua** [comment]
@mauritsvanrees this fix also can be enabled for **1.7-maintenance** branch.

**Comment by mauritsvanrees** [comment]
I already made pull requests for both branches. :-)
- 1.7: #180 
- master: #181

**Comment by macagua** [comment]
@mauritsvanrees really thanks you! 

I waiting your merge :-D

**Comment by mauritsvanrees** [comment]
Steve has merged and released.
The fix is in 1.7.20 and 1.8.2.

**Comment by macagua** [comment]
@mauritsvanrees @smcmahon thanks you!
